src/main/java/at/hannibal2/skyhanni/data/BannedClasses.kt

```diff
@@ -0,0 +1,20 @@
+package at.hannibal2.skyhanni.data
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.jsonobjects.repo.BannedClassesJson
+import at.hannibal2.skyhanni.events.RepositoryReloadEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+object BannedClasses {
+
+    private var list = listOf<String>()
```

Comment:
use a set instead of a list. The set is about ~25% faster.
<details>
<summary>Test reuslt</summary>
Top being list and bottom being set

![image](https://github.com/hannibal002/SkyHanni/assets/85900443/2ab8ef25-9940-41f7-82e2-89dc644aa9ce)
![image](https://github.com/hannibal002/SkyHanni/assets/85900443/b917f102-4af1-4be3-af45-cd53af11a0bc)
</details>
<details>
<summary>Code</summary>

Tested twice once with list first and once with set first. Couldn't test on the same string since the compiler optimized my test away when timing both for the same string.
```Kotlin
package at.hannibal2.skyhanni.data

import at.hannibal2.skyhanni.SkyHanniMod
import at.hannibal2.skyhanni.data.jsonobjects.repo.BannedClassesJson
import at.hannibal2.skyhanni.events.RepositoryReloadEvent
import at.hannibal2.skyhanni.utils.CollectionUtils.editCopy
import at.hannibal2.skyhanni.utils.LorenzDebug
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import kotlin.time.Duration
import kotlin.time.measureTime

object BannedClasses {

    private var list = listOf<String>()
    private var set = setOf<String>()

    private var listTime = Duration.ZERO
    private var setTime = Duration.ZERO

    private var ab = false

    @SubscribeEvent
    fun onRepoReload(event: RepositoryReloadEvent) {
        list = event.getConstant<BannedClassesJson>("BannedClasses").bannedClasses[SkyHanniMod.version] ?: emptyList()
        set = list.editCopy { add(0, "") }.drop(1).toSet()
    }

    fun isBanned(provider: () -> Class<*>?): Boolean = list.isNotEmpty() && isBanned(provider())

    fun isBanned(javaClass: Class<*>?): Boolean {
        val result: Boolean
        val name = javaClass?.name ?: return false
        if (ab) {
            listTime += measureTime {
                result = list.contains(name)
            }
        } else {
            setTime += measureTime {
                result = set.contains(name)
            }
        }
        ab = !ab
        LorenzDebug.log("Times:\n$listTime\n$setTime")
        return result
    }
}
```
</details>